core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -289,6 +324,14 @@ public Path pidFile() {
     }
 
     /** Path to the default temp directory used by the JDK */
+    public Path systemTmpFile() {
```

Comment:
do we really need this or would the tmpFile be enough?

Reply:
The only place it's used is in the security manager setup so that the security manager allows access to `java.io.tmpdir`, not just the new sub-directory below it.  The reason I thought this would be a good idea is that there could be a 3rd party Elasticsearch plugin that uses a Java library that assumes it can use `java.io.tmpdir`.  At the moment this would work, but if we changed our security manager rules to disallow access to `java.io.tmpdir` then any such plugins would stop working.

I can achieve this differently so that `java.io.tmpdir` is not made available by the `Environment` class, but the security manager still allows access to it for the benefit of Java libraries used by 3rd party plugins.  Or I can just leave the security manager setup as it was so that _everything_ that wants a temp directory has to use `Environment.tmpFile()`.  Do you have a preference?

Reply:
can't you just use the system property in the security manager instead?